### PR TITLE
Surface System.Runtime.Caching content

### LINKED ISSUE #8478: System.Runtime.Caching?
Why use this asp.net specific caching mechanism over System.Runtime.Caching?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 13f5f8f9-18dd-9c12-9ed1-fcc6b44ba7fd
* Version Independent ID: 640b4f36-5886-7b69-1c95-cfc8574a02fd
* Content: [Cache in-memory in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/memory?view=aspnetcore-2.1)
* Content Source: [aspnetcore/performance/caching/memory.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/performance/caching/memory.md)
* Product: **aspnet-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #8478 

[Internal Review Topic (links directly to section)](https://review.docs.microsoft.com/en-us/aspnet/core/performance/caching/memory?view=aspnetcore-1.0&branch=pr-en-us-8550#systemruntimecachingmemorycache)

* Went with a section over a NOTE, as this content seems very relevant/important to lib/app porting.
* The topic could use a UE pass, but 🏃 right now. It's scheduled for more work on #6808.

### CODE REVIEW

**Comment by guardrex** [comment]
@scottaddie This one, too. See if this is enough for the update :point_right: https://github.com/aspnet/Docs/issues/8478#issuecomment-421524172

File: aspnetcore/performance/caching/memory.md [hunk 1]
```diff
@@ -25,7 +25,15 @@ The `IMemoryCache` cache will evict cache entries under memory pressure unless t
 
 The in-memory cache can store any object; the distributed cache interface is limited to `byte[]`.
 
-### Cache guidelines
+## System.Runtime.Caching/MemoryCache
+
+<xref:System.Runtime.Caching>/<xref:System.Runtime.Caching.MemoryCache> ([NuGet package](https://www.nuget.org/packages/System.Runtime.Caching/)) is available for use in .NET Standard 2.0 or later libraries and implementations of .NET, including apps that target ASP.NET Core 2.0 or later and .NET Framework 4.5 or later.
```
Comment (scottaddie) [inline aspnetcore/performance/caching/memory.md]: This sentence reads funny to me. Could we say something like the following instead?

...can be used with:

* .NET Standard 2.0 or later
* any [.NET implementation](https://docs.microsoft.com/dotnet/standard/net-standard#net-implementation-support) targeting .NET Standard 2.0 or later

For example, ASP.NET Core 2.0 apps targeting .NET Core or .NET Framework 4.6.1 or later."

File: aspnetcore/performance/caching/memory.md [hunk 2]
```diff
@@ -25,7 +25,15 @@ The `IMemoryCache` cache will evict cache entries under memory pressure unless t
...
+
+[Microsoft.Extensions.Caching.Memory](https://www.nuget.org/packages/Microsoft.Extensions.Caching.Memory/)/`IMemoryCache` (described in this topic) is recommended over `System.Runtime.Caching`/`MemoryCache` because it's better integrated into ASP.NET Core (for example, `IMemoryCache` works natively with ASP.NET Core [dependency injection](xref:fundamentals/dependency-injection)).
```
Comment (scottaddie) [inline aspnetcore/performance/caching/memory.md]: The "for example" part should be the start of a new sentence.

File: aspnetcore/performance/caching/memory.md [hunk 3]
```diff
@@ -25,7 +25,15 @@ The `IMemoryCache` cache will evict cache entries under memory pressure unless t
...
+
+Use `System.Runtime.Caching`/`MemoryCache` as a compatibility bridge when porting code to ASP.NET Core.
```
Comment (scottaddie) [inline aspnetcore/performance/caching/memory.md]: I assume this refers to porting from ASP.NET 4.x. Let's specify that here.

**Comment by guardrex** [comment]
Build system shot me an email with some errors ...

Error | Unspecified |   |   | File:../aspnetcore/mvc/razor-pages/razor-pages-convention-features.md not existed. |   | 09/18/2018 21:17:31
Error | Unspecified |   |   | Some error ocurred while parsing changelist file: W:\gxtu-log-41afaf87\change-list-with-redirection.tsv. |   | 09/18/2018 21:17:31
Error | Unspecified |   |   | File:mvc/razor-pages/razor-pages-convention-features.md not existed. |   | 09/18/2018 21:17:34
Error | Unspecified |   |   | Some error ocurred while parsing changelist file: W:\gxtu-log-41afaf87\change-list-with-redirection.tsv. |   | 09/18/2018 21:17:34

Idk where it thinks those are.

**Pull request merged**